frontend/src/container/QueryBuilder/filters/QueryBuilderSearchV2/QueryBuilderSearchV2.tsx

```diff
@@ -0,0 +1,857 @@
+			? value?.trim()?.replace(/,\s*$/, '')
+			: value?.trim();
+
+		const indexInQueryTags = queryTags.findIndex((qTag) => isEqual(qTag, value));
+		const tagDetails = tags[indexInQueryTags];
+
+		const onCloseHandler = (): void => {
+			onClose();
+			setSearchValue('');
+			setTags((prev) => prev.filter((t) => !isEqual(t, tagDetails)));
+		};
+
+		const tagEditHandler = (value: string): void => {
+			setCurrentFilterItem(tagDetails);
+			setSearchValue(value);
+			setCurrentState(DropdownState.ATTRIBUTE_VALUE);
+			setTags((prev) => prev.filter((t) => !isEqual(t, tagDetails)));
+		};
+
+		const isDisabled = !!searchValue;
+
+		return (
+			<Tag
+				closable={!searchValue && closable}
+				onClose={onCloseHandler}
+				className={tagDetails?.key?.type || ''}
+			>
+				<Tooltip title={chipValue}>
+					<TypographyText
+						ellipsis
+						$isInNin={isInNin}
+						disabled={isDisabled}
+						$isEnabled={!!searchValue}
+						onClick={(): void => {
+							if (!isDisabled) tagEditHandler(value);
+						}}
+					>
+						{chipValue}
+					</TypographyText>
+				</Tooltip>
+			</Tag>
+		);
+	};
+
+	return (
+		<div className="query-builder-search-v2">
+			<Select
+				ref={selectRef}
+				getPopupContainer={popupContainer}
+				virtual={false}
+				showSearch
+				tagRender={onTagRender}
+				transitionName=""
+				choiceTransitionName=""
+				filterOption={false}
+				open={isOpen}
+				suffixIcon={
+					// eslint-disable-next-line no-nested-ternary
+					!isUndefined(suffixIcon) ? (
+						suffixIcon
+					) : isOpen ? (
+						<ChevronUp size={14} />
+					) : (
+						<ChevronDown size={14} />
+					)
+				}
+				onDropdownVisibleChange={setIsOpen}
+				autoClearSearchValue={false}
+				mode="multiple"
+				placeholder={placeholder}
+				value={queryTags}
+				searchValue={searchValue}
+				className={cx(
+					!currentFilterItem?.key && !showAllFilters && dropdownOptions.length > 3
+						? 'show-all-filters'
+						: '',
+					className,
+				)}
+				rootClassName="query-builder-search"
+				disabled={isMetricsDataSource && !query.aggregateAttribute.key}
+				style={selectStyle}
+				onSearch={handleSearch}
+				onSelect={handleDropdownSelect}
+				onInputKeyDown={onInputKeyDownHandler}
+				notFoundContent={loading ? <Spin size="small" /> : null}
+				showAction={['focus']}
+				onBlur={handleOnBlur}
+				// eslint-disable-next-line react/no-unstable-nested-components
+				dropdownRender={(menu): ReactElement => (
+					<CustomDropdown
+						menu={menu}
+						options={dropdownOptions}
+						onChange={(val: TagFilter): void => {
+							setTags((prev) => [...prev, ...(val.items as ITag[])]);
+						}}
+						searchValue={searchValue}
+						exampleQueries={suggestionsData?.payload?.example_queries || []}
+						tags={tags}
+						currentFilterItem={currentFilterItem}
+					/>
+				)}
+			>
+				{dropdownOptions.map((option) => (
+					<Select.Option
+						key={isObject(option.value) ? JSON.stringify(option.value) : option.value}
```

Comment:
Move to a variable and use it in the 2 places

Reply:
done